lib/iuliia/schema.rb

```diff
@@ -2,52 +2,35 @@
 
 module Iuliia
   module Schema
-    extend self
-
-    SCHEMA = Struct.new(
-      :description,
-      :url,
-      :mapping,
-      :prev_mapping,
-      :next_mapping,
-      :ending_mapping
-    )
-
-    def schema(schema_name)
-      schemas[schema_name]
-    end
+    class << self
 
-    def available_schemas
-      schemas.map do |schema_name, schema|
-        [schema_name, schema.description]
+      def [](schema_name)
+        schemas[schema_name]
       end
-    end
 
-    private
+      alias_method :schema, :[]
 
-    def schemas
-      @schemas ||= load_schemas
-    end
+      def available_schemas
+        load_schemas.transform_values(&:description).to_a
+      end
 
-    def load_schemas
-      Dir['lib/schemas/*.json'].map do |schema_file|
-        file = File.read(schema_file)
-        parse_and_create_schema(file)
-      end.reduce({}, :update)
-    end
+      private
 
-    def parse_and_create_schema(file)
-      parsed_data = JSON.parse(file)
-      schema = SCHEMA.new(
-        parsed_data['description'],
-        parsed_data['url'],
-        parsed_data['mapping'],
-        parsed_data['prev_mapping'],
-        parsed_data['next_mapping'],
-        parsed_data['ending_mapping']
-      )
-
-      { parsed_data['name'] => schema }
+      def schemas
+        @schemas ||= Hash.new { |h, k| h[k] = load_schema(k) }
```

Comment:
Use [`Hash#default_proc`](https://docs.ruby-lang.org/en/2.7.0/Hash.html#method-i-default_proc) for lazy schemas loading.